File: backend/app/app/schemas/user_feedback.py
from pydantic import BaseModel
from datetime import datetime
from datetime import datetime

from pydantic import BaseModel


class UserFeedbackBase(BaseModel):
    feedback_given_user_id: int = None
    feedback_received_user_id: int = None
    project_worker_id: int = None
    rating: int = None
    comment: str = None
    created_dt: datetime = None
    
    
class UserFeedbackCreate(UserFeedbackBase):
    class Config:
        orm_mode = True


class UserFeedbackUpdate(UserFeedbackBase):
    id: int = None
    
    class Config:
        orm_mode = True


class UserFeedbackWithProject(UserFeedbackBase):
    project_worker: "project_worker.ProjectWorkerProject"

    class Config:
        orm_mode = True


class UserFeedback(UserFeedbackUpdate):
    pass


from . import project_worker

UserFeedbackWithProject.update_forward_refs()

File: frontend/src/interfaces/UserFeedback.ts
import { IProjectWorkerUserInfo } from '@/interfaces'

export interface IUserFeedback {
  rating: number;
  comment: string;
}

export interface IUserFeedbackCreate {
  feedback_received_user_id: number;
  project_worker_id: number;
  rating: number;
  comment: string;
}

export interface IUserFeedbackInfo {
  feedback_received_user_id: number;
  rating: number;
  comment: string;
  project_worker: IProjectWorkerUserInfo
  created_dt: Date;
}